pkg/snap/util/dqlite_test.go

```diff
@@ -84,3 +85,80 @@ Role: 0`,
 	})
 
 }
+
+func TestUpdateDqliteBindAddress(t *testing.T) {
+	findMatchingBindAddressMock := func(addr string) (string, error) {
+		return "10.10.10.10", nil
+	}
+
+	t.Run("MustFailIfNodeAlreadyKnown", func(t *testing.T) {
+
+		s := &mock.Snap{
+			DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0`,
+			DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+		}
+
+		g := NewWithT(t)
+		err := snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1", "127.0.0.1", findMatchingBindAddressMock)
+		g.Expect(err).To(MatchError("the joining node (127.0.0.1) is already known to dqlite"))
+
+	})
+
+	t.Run("MustNotUpdateIfMultipleNodes", func(t *testing.T) {
+		s := &mock.Snap{
+			DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0
+- Address: 10.10.10.10:19001
+  ID: 12345678987654321
+  Role: 0`,
+			DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+		}
+
+		g := NewWithT(t)
+		err := snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1:19001", "8.8.8.8", findMatchingBindAddressMock)
+		g.Expect(err).To(BeNil())
+		g.Expect(s.WriteDqliteUpdateYamlCalledWith).To(BeEmpty())
+	})
+
+	t.Run("MustUpdateIfOneNodeAndBindsLocal", func(t *testing.T) {
+		s := &mock.Snap{
+			DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0`,
+			DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+		}
+		// Update cluster yaml asynchronously
+		// The mock snap implementation doesn't actually write the yaml but instead
+		// sets `WriteDqliteUpdateYamlCalledWith`.
+		// Thus, we need to manually change the config as otherwise `WaitForDqliteCluster`
+		// would wait forever.
```

Comment:
What if it did? Should we just perhaps update the mock implementation?

Otherwise it feels like we are not testing something significant.

Reply:
TBH I was also a bit confused on why the mock implementation is setup that way. I assumed there was a historic reason for that as I don't see a real reason for that implementation. However, I didn't want to mix concerns here and change that implementation in this PR.

Also:
1. I don't see how actually writing the yaml file in the mock changes the test behavior. We would get rid of this async manual change (which arguably is very ugly) but apart from that, there is not a real change, is it?
2. The test verifies the branching and call logic of the function. What else would you like to test in this unittest?